tests/test.cpp

```diff
@@ -105,6 +105,24 @@ static void testFeature() {
     feature<double> pf { point<double>() };
     assert(pf.geometry.is<point<double>>());
     assert(pf.properties.size() == 0);
+
+    auto &p = pf.properties;
+
+    p["bool"] = true;
+    p["string"] = "foo";
+    p["double"] = 2.5;
+    p["uint"] = 10;
+    p["int"] = -10;
+    p["null"] = nullptr;
+
+    assert(p["bool"] == true);
+    assert(p["string"] == "foo");
```

Comment:
The assertions for string, uint, and int are passing only by accident due to unexpected implicit `bool` conversions. I'll add a commit to show this.

Upstream issue is https://github.com/mapbox/variant/issues/100.